Yosemite/Yosemite/Stores/ProductAttributeTermStore.swift

```diff
@@ -0,0 +1,175 @@
+import Foundation
+import Networking
+import Storage
+
+public final class ProductAttributeTermStore: Store {
+    private let remote: ProductAttributeTermRemote
+
+    private lazy var sharedDerivedStorage: StorageType = {
+        return storageManager.newDerivedStorage()
+    }()
+
+    public override init(dispatcher: Dispatcher, storageManager: StorageManagerType, network: Network) {
+        self.remote = ProductAttributeTermRemote(network: network)
+        super.init(dispatcher: dispatcher, storageManager: storageManager, network: network)
+    }
+
+    override public func registerSupportedActions(in dispatcher: Dispatcher) {
+        dispatcher.register(processor: self, for: ProductAttributeTermAction.self)
+    }
+
+    override public func onAction(_ action: Action) {
+        guard let action = action as? ProductAttributeTermAction else {
+            assertionFailure("ProductAttributeTermStore received an unsupported action")
+            return
+        }
+
+        switch action {
+        case let .synchronizeProductAttributeTerms(siteID, attributeID, onCompletion):
+            synchronizeAllProductAttributeTerms(siteID: siteID, attributeID: attributeID, fromPageNumber: 0) { error in
```

Comment:
Alternatively, since our use case in the app layer is to sync **all** attribute terms, we don't need the `fromPageNumber` param and can rename this action to be like `synchronizeAllProductAttributeTerms`? Unless we are supporting syncing a subset of pages?

Reply:
Oh nice catch, I've updated to use page 1 as the first page 🤦.
We do need the `fromPageNumber` and the `fromPageNumber + 1` because that is the parameter that changes on the recursive call in order to fetch the next page when needed!